master/buildbot/steps/source/bzr.py

```diff
@@ -50,7 +50,9 @@ def __init__(self, repourl=None, baseURL=None, mode='incremental',
         if baseURL is not None and defaultBranch is None:
             raise ValueError("you must provide defaultBranch with baseURL")
 
-        assert self.mode in ['incremental', 'full']
+        if not self._hasAttrGroupMember('mode', self.mode):
```

Comment:
I don't have a problem with it but have a few concerns about this.

1. This would need to be a method that gets called. The p4 step has an extra check on the if and the repo step is completely different.
2. It looks like most of the steps are slightly different in their implementations, especially error reporting. Moving this check to the base class would probably entail updating all of the source steps to be homogeneous. And I have no idea what the recommended way is
3. The same unification could apply to the `method` parameter too. My thought was make `mode` consistent, then `method`, then error reporting.

Expanding on point 2, I see three ways right now: raising `ValueError`, `ConfigError(errors)`, and `buildbot.config.error(error)`. There's also a few `assert`s. Any recommendations? (Bring in @djmitche too).